lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,43 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
+            else
+              self._value = value
+            end
           end
 
           def href(value)
             self._href = value
+            nil
           end
 
           def meta(value)
             self._meta = value
+            nil
           end
 
+          def serializable_hash
```

Comment:
`as_json` is the way to go IMO (for the record, `as_json` means build something ready to be directly transformed into JSON, i.e. a Hash mainly, but we could extend the meaning to be "something that will eventually be part of a Hash representing a JSON document", and `to_json` means actually building a JSON string from the object).